index.js

```diff
@@ -83,51 +81,32 @@ const nonWindowsSingleCall = async (options = {}) => {
 	});
 
 	const lines = stdout.trim().split('\n');
+	const COMMAND_POSITION = lines[0].lastIndexOf('COMMAND') - lines[0].indexOf('COMMAND');
 	lines.shift();
 
-	let psIndex;
-	let commPosition;
-	let argsPosition;
-
-	const processes = lines.map((line, index) => {
+	return lines.map(line => {
 		const match = psOutputRegex.exec(line);
 		if (match === null) {
 			throw new Error(ERROR_MESSAGE_PARSING_FAILED);
 		}
 
-		const {pid, ppid, uid, cpu, memory} = match.groups;
+		const {pid, ppid, uid, cpu, memory, commAndArgs} = match.groups;
+		const comm = commAndArgs.slice(0, COMMAND_POSITION).trim();
+		if (comm === 'ps') {
+			return null;
+		}
 
-		const processInfo = {
+		const args = commAndArgs.slice(COMMAND_POSITION);
+		return {
 			pid: Number.parseInt(pid, 10),
 			ppid: Number.parseInt(ppid, 10),
 			uid: Number.parseInt(uid, 10),
 			cpu: Number.parseFloat(cpu),
 			memory: Number.parseFloat(memory),
-			name: undefined,
-			cmd: undefined
+			name: comm,
+			cmd: args
 		};
-
-		if (processInfo.pid === psPid) {
-			psIndex = index;
-			commPosition = line.indexOf('ps', match[0].length);
-			argsPosition = line.indexOf('ps', commPosition + 2);
-		}
-
-		return processInfo;
-	});
-
-	if (psIndex === undefined || commPosition === -1 || argsPosition === -1) {
-		throw new Error(ERROR_MESSAGE_PARSING_FAILED);
-	}
-
-	const commLength = argsPosition - commPosition;
-	for (const [index, line] of lines.entries()) {
-		processes[index].name = line.slice(commPosition, commPosition + commLength).trim();
-		processes[index].cmd = line.slice(argsPosition).trim();
-	}
-
-	processes.splice(psIndex, 1);
-	return processes;
+	}).filter(line => line);
```

Comment:
I don't remember what exactly @sindresorhus thinks of such things, but for me it is extremely bad style, this line makes me think all of the lines were boolean instead of strings and reread all the code from the start to figure out what I missed.